### DISCUSSION TITLE
New virtual machine never starts

### DISCUSSION THREAD

**Discussion by hannesa2** [utmapp/UTM#3682] [UTM Help (macOS)]
A new created virtual machine never starts
<img width="439" alt="image" src="https://user-images.githubusercontent.com/3314607/155663617-7f7a2674-c209-4920-8266-0047543a3cb8.png">

<img width="638" alt="image" src="https://user-images.githubusercontent.com/3314607/155663654-196146c2-a15f-4a33-8d85-44acf9290377.png">

**Configuration**
* UTM Version: 3.1.1 (48)
* OS Version: macOS 12.2.1
* Apple Silicon

**Crash log**
There is no crash

**Debug log**
[debug.log](https://github.com/utmapp/UTM/files/8138928/debug.log)

These are the last entries 
```
qemu-aarch64-softmmu: warning: Spice: playback:0 (0x14b8c7940): setsockopt failed, Operation not supported on socket
qemu-aarch64-softmmu: warning: Spice: record:0 (0x14b8c79f0): setsockopt failed, Operation not supported on socket
```

**Upload VM**
[UTM](http://dev.mxtracks.info/ubuntu18.utm.zip)

**Comment by osy** [comment]
Follow https://mac.getutm.app/gallery/ubuntu-20-04

**Reply by hannesa2** [reply]
Sorry, I can't use `20.04`, AOSP needs `18.04`

**Reply by osy** [reply]
You can use https://cdimage.ubuntu.com/releases/18.04/release/ with the same instructions

**Reply by hannesa2** [reply]
Before I run successful a 18.04, but my harddrive runs out of disk space.
-> Because disk wasn't extendable I added an additional one
-> with additional disk, I lost network interface -wtf-
-> remove additional disk, I got network interface back
-> to create an issue in this repository, I replay these steps, to _document_ it. I added an additional one
-> with additional disk, I lost network interface again
-> remove additional disk, my virtual image freeze during boot -wtf- (I replayed this unsuccessfully more then 3 times again)

So I gave up and created a new 18.04 .... now I'm here

**Reply by hannesa2** [reply]
> You can use https://cdimage.ubuntu.com/releases/18.04/release/ with the same instructions

This is exactly what I did

**Reply by hannesa2** [reply]
Please have a look on my screen cast what I did
https://user-images.githubusercontent.com/3314607/155674057-4cd19615-8915-477c-89aa-484c4f0ed8ce.mov

**Reply by brnuts** [reply]
It is also happening on my 20.04 Ubuntu. The last debug message is:
"qemu-aarch64-softmmu: warning: Spice: record:0 (0x12986b9f0): setsockopt failed, Operation not supported on socket"

Difficult to investigate as DEBUG is not showing any error, but a warning.

**Reply by brnuts** [reply]
The root cause is the kernel.

Look here: https://github.com/utmapp/UTM/issues/2682
and: https://gitlab.com/qemu-project/qemu/-/issues/899

It is Linux 5.4.0-104. Here is a temporary workaround (skip steps 1-4 if the VM worked previously and only stopped booting recently):

- Open the VM settings and under QEMU -> Tweaks, uncheck "Use Hypervisor"
- Boot into Linux, but it will be VERY slow because it is using emulation instead of virtualization.
- Once inside, install the last version of Linux sudo apt install linux-image-5.4.0-100-generic
- Shut down the VM, open VM settings and under QEMU -> Teaks, check "Use Hypervisor" to re-enable virtualization.
- Start the VM and when the UTM logo shows up, hold "Shift" to enter the GRUB menu
- Select "Advanced Options for Ubuntu"
- Select "Ubuntu, with Linux 5.4.0-100-generic" (or anything lower than 5.4.0-104)

You can now boot into Ubuntu again. You may choose to uninstall 5.4.0-104 and temporarily mask it from being updated in the future.

**Reply by hannesa2** [reply]
❤️ I'm not alone anymore. UTM is complete broken and no more usage at all.
Again, this is no discussion, it's a blocker bug !

**Reply by brnuts** [reply]
Have u tried the 5.4.0-100 kernel from Grub? Mine is working now.

**Reply by hannesa2** [reply]
Sadly I've been waited for 5 minutes without any progress 
<img width="353" alt="image" src="https://user-images.githubusercontent.com/3314607/158534525-607750cc-6a96-4dce-a60b-d3c7e915d6ec.png">


<img width="574" alt="image" src="https://user-images.githubusercontent.com/3314607/158534438-9f1adabf-34a1-4944-8a2c-5d76dd4321e2.png">

**Comment by hannesa2** [comment]
For me it looks like to be not a discussion, it's a bug !

**Reply by brnuts** [reply]
It is also happening on Ubuntu 20.04

**Comment by mfractal** [comment]
did you ever solve this ? i am getting the exact same issue installing ubuntu 22 arm. will try 18 now to see if it works.

**Reply by hannesa2** [reply]
Yes, it works now. But sadly I don't know exactly what was wrong on my side. As far as I remember, first I created the VM by changing as less as possible compared to the defaults. -> This was wrong
I guess I changed one (?) value, then it was working. 

But I don't use it often, because I spent too much time on blank screens and nobody takes care about this.

**Comment by moivica** [comment]
I have the same problem with kali linux - but the difference is that my machine was running over a year, and suddenly stopped - i have the same errors in logs  like in the report: 

```
qemu-aarch64-softmmu: warning: Spice: playback:0 (0x1360d4140): setsockopt failed, Operation not supported on socket
qemu-aarch64-softmmu: warning: Spice: record:0 (0x1360d41e0): setsockopt failed, Operation not supported on socket
```

Is there a way to save the data from the virtual machine to local drive if the virtual machine cannot be started again ?

**Comment by n4chh** [comment]
Does any one got any news  about how to solve this? I got a manjaro vm and after upgrading system after installation and reboot, does display those messages:

```
qemu-aarch64-softmmu: warning: Spice: playback:0 (0x14c0db140): setsockopt failed, Operation not supported on socket
qemu-aarch64-softmmu: warning: Spice: record:0 (0x14c0db1e0): setsockopt failed, Operation not supported on socket
```
I search through the internet and try to launch the machine directly with qemu but nothing works
